src/Makefile.am

```diff
@@ -99,7 +99,10 @@ BITCOIN_CORE_H = \
   compressor.h \
   consensus/consensus.h \
   consensus/params.h \
+  consensus/blockruleindex.h \
```

Comment:
We're avoiding state-full structures like this on consensus folder. Even if the folder has to depend on state-full structures like CBlockIndex, CCoinsViewCache and now also BlockRuleIndex temporarily, the consensus code will have to be decoupled from them to complete libconsensus, so chain.o, coins.o and blockruleindex.o don't belong in this folder.